src/App.tsx

```diff
@@ -1,26 +1,144 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { addTodos, deleteTodos, getTodos, USER_ID } from './api/todos';
+import { Header } from './component/Header';
+import { Section } from './component/Section/Section';
+import { Footer } from './component/Footer/Footer';
+import { Todo } from './types/Todo';
+import { Error } from './component/Error';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessege, setErrorMessege] = useState<string | null>(null);
+  const [newTodo, setNewTodo] = useState<string>('');
+  const [loading, setLoading] = useState(false);
+  const [selectedFilter, setselectedFilter] = useState<string>(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingId, setloadingId] = useState<number[]>([]);
+
+  const loadTodos = async (): Promise<void> => {
+    setLoading(true);
+    try {
+      setTodos(await getTodos());
+    } catch {
+      setErrorMessege('Unable to load todos');
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  async function handleAddTodo(event: React.FormEvent) {
+    event.preventDefault();
+    if (!newTodo.trim()) {
+      setErrorMessege('Title should not be empty');
+
+      return;
+    }
+
+    setLoading(true);
+
+    const addTodo = {
+      id: todos.length + 1,
+      userId: USER_ID,
+      title: newTodo.trim(),
+      completed: false,
+    };
+
+    try {
+      setTempTodo(addTodo);
+      setTodos([...todos, await addTodos(addTodo)]);
+      setNewTodo('');
+    } catch {
+      setErrorMessege('Unable to add a todo');
+    } finally {
+      setTempTodo(null);
+      setLoading(false);
+    }
+  }
+
+  async function handleDeleteTodo(id: number) {
```

Comment:
It is important to stick to one way of creating functions, fix them everywhere
```suggestion
  const handleDeleteTodo = async (id: number) => {
```